src/core.c/Sequence.pm6

```diff
@@ -28,8 +28,13 @@ my role PositionalBindFailover {
           ($!list := List.from-iterator(self.iterator))
         )
     }
+
     multi method list(::?CLASS:D:) {
-        List.from-iterator(self.iterator)
+        nqp::if(
+          nqp::isconcrete($!list),
+          $!list,
+          List.from-iterator(self.iterator)
```

Comment:
Shouldn't that need to set `$!list` ?

Reply:
No, that's what `.cache` does.

Reply:
Correct, only `.cache` should set the `list`, this is just using the cached list if it exists.

Reply:
(Which I think we can safely get away with.)